contractcourt/channel_arbitrator.go

```diff
@@ -1050,6 +1050,20 @@ func (c *ChannelArbitrator) stateStep(
 		if err != nil {
 			log.Errorf("ChannelArbitrator(%v): unable to "+
 				"force close: %v", c.cfg.ChanPoint, err)
+
+			// We tried to force close (HTLC may be expiring from
+			// our PoV, etc), but we think we've lost data. In this
+			// case, we'll not force close, but terminate the state
+			// machine here to wait to see what confirms on chain.
+			if errors.Is(err, lnwallet.ErrForceCloseLocalDataLoss) {
+				log.Error("ChannelArbitrator(%v): broadcast "+
+					"failed due to local data loss, "+
+					"waiting for on chain confimation...",
+					c.cfg.ChanPoint)
+
+				return StateCommitmentBroadcasted, nil, nil
```

Comment:
not sure about this step. Since we failed to create and broadcast the FC tx here, it means we should abort the `stateStep` as the arbitrator will now have undefined behaviors when it tries to create resolutions for under this state?

Reply:
I agree, I think we should progress only if the channel is marked with `ChanStatusLocalDataLoss` and the trigger is either `remoteCloseTrigger` or `breachCloseTrigger` (covering our bases)

Reply:
Yeah so my mental model is that eventually something will confirm, and we'll get a signal from the chain watcher that the channel has been force closed. Moving to the next state also means we won't try to broadcast again on start up either. We'll instead get the `StateContractClosed` signal. The state will be "unknown" so we'll pass in an empty commitment, like the SCB flow, then just play the commitment outputs. 

The addition of the `ErrForceCloseLocalDataLoss` error means we'll only ever do this if we had local data loss, so I think that's covered?